### PR TITLE
Add support for `buildah run --hostname`

### PR DESCRIPTION
**Pull request by rhatdan**
Need to set the hostname inside of a container.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes https://github.com/projectatomic/buildah/issues/265

**Comment by rhatdan** [comment]
bot retest

**Comment by rhatdan** [comment]
bot retry

**Comment by rhatdan** [comment]
@jlebon Any idea what is going wrong?

**Comment by TomSweeneyRedHat** [comment]
The change LGTM, wonder what's up with Red Hat CI.

**Comment by jlebon** [comment]
bot, retest this please

**Comment by jlebon** [comment]
It should be fine now... hopefully. :)

File: cmd/buildah/run.go [hunk 1]
```diff
@@ -16,6 +16,11 @@ import (
 var (
 	runFlags = []cli.Flag{
 		cli.StringFlag{
+			Name:  "hostname",
+			Usage: "Set the hostname inside of the container",
+			Value: "localhost",
```
Comment (nalind) [inline cmd/buildah/run.go]: A default can be provided in an image, so I prefer to not override it unless we're explicitly asked to.

**Comment by nalind** [comment]
This needs a test.

File: cmd/buildah/run.go (hunk 1 above)
Comment (rhatdan) [reply cmd/buildah/run.go]: ok

**Comment by rhatdan** [comment]
@nalind has a test and passes.

**Comment by nalind** [comment]
LGTM.
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
Looks like this hung.

@rh-atomic-bot retry

**Comment by rhatdan** [comment]
@jlebon Looks like I have hung up the test bot, any chance you could clear it.

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
@jlebon This was timing out all weekend.

**Pull request closed without merging**